crates/tabby/src/serve/health.rs

```diff
@@ -1,14 +1,41 @@
-use std::sync::Arc;
+use std::{env::consts::ARCH, sync::Arc};
 
 use axum::{extract::State, Json};
 use serde::{Deserialize, Serialize};
+use sysinfo::{CpuExt, System, SystemExt};
 use utoipa::ToSchema;
 
+#[derive(Serialize, Deserialize, ToSchema, Clone, Debug)]
+struct CPUInfo {
+    brand: String,
+    number: usize,
+}
+
+impl CPUInfo {
+    pub fn new() -> Self {
+        let mut sys = System::new_all();
+        sys.refresh_cpu();
+        let cpus = sys.cpus();
+        let brand = if cpus.len() > 0 {
+            let cpu = &cpus[0];
+            cpu.brand().to_string()
+        } else {
+            "no cpus assigned".to_string()
```

Comment:
is it really possible a computer doesn't have cpu?

Reply:
Maybe not, but this part of code is error proof so that even if `sys.cpus()` somehow returns some empty array (maybe because of some instance virtualization breaking the library? I don't know), the `[0]` operation won't crash our tabby binary.